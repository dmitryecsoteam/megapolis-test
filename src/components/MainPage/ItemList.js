import React from 'react';
import Item from './Item';
import PropTypes from 'prop-types';

/**
 * 1. Если массив items пустой, то рендерим сообщение, что элементов нет
 * 2. Если размер массива больше нуля, то проверяем первый элемент. Если он не равен 'INIT' (начальное состояние redux store),
 *    то значит данные из апи подгрузились и можно их рендерить с помощью map
 */
const ItemList = ({ items, deleteItem }) => (
    <div className='item-list__container'>
        {items.length === 0 ? <div>н/д</div> :
            items[0] !== 'INIT' && items.map(item => (
                <Item key={item.id} {...item} deleteItem={deleteItem} />
            ))}
    </div>
);

ItemList.propTypes = {
    items: PropTypes.array.isRequired,
    deleteItem: PropTypes.func.isRequired
}

export default ItemList;