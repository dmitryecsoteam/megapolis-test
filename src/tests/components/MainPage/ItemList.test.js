import React from 'react';
import { shallow } from 'enzyme';
import ItemList from '../../../components/MainPage/ItemList';

test('should not render list, if items array contains "INIT"', () => {
    const wrapper = shallow(<ItemList items={['INIT']} deleteItem={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render message if items array is empty', () => {
    const wrapper = shallow(<ItemList items={[]} deleteItem={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render list of items from items array', () => {
    const wrapper = shallow(<ItemList items={[{ id: 1, title: 'first' }, { id: 2, title: 'second' }]} deleteItem={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
});