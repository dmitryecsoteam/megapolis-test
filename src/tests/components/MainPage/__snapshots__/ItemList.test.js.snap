// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render list, if items array contains "INIT" 1`] = `
<div
  className="item-list__container"
/>
`;

exports[`should render list of items from items array 1`] = `
<div
  className="item-list__container"
>
  <Item
    deleteItem={[Function]}
    id={1}
    key="1"
    title="first"
  />
  <Item
    deleteItem={[Function]}
    id={2}
    key="2"
    title="second"
  />
</div>
`;

exports[`should render message if items array is empty 1`] = `
<div
  className="item-list__container"
>
  <div>
    н/д
  </div>
</div>
`;
